README.md

```diff
@@ -214,19 +206,24 @@ The environment variables will be encrypted in the client-side and decrypted in
 
 ### Access the App
 
-Once the app is deployed and listening on an HTTP port, you can access the HTTP service via dstack-gateway's public domain. The ingress mapping rules are:
+Once your app is deployed and listening on an HTTP port, you can access it through dstack-gateway's public domain using these ingress mapping rules:
+
+- `<id>-[<port>][s|g].<base_domain>` → maps to port `<port>` in the CVM
+
+**Examples**:
 
-- `<id>[s].<base_domain>` maps to port `80` or `443` if with `s` in the CVM
-- `<id>-<port>[s].<base_domain>` maps to port `<port>` in the CVM
+- `3327603e03f5bd1f830812ca4a789277fc31f577-8080.app.kvin.wang` → port `8080` (HTTP/1.1)
+- `3327603e03f5bd1f830812ca4a789277fc31f577-8080g.app.kvin.wang` → port `8080` (HTTPS/2 with TLS termination)
+- `3327603e03f5bd1f830812ca4a789277fc31f577-8080s.app.kvin.wang` → port `8080` (TLS passthrough)
 
-**Example**: `3327603e03f5bd1f830812ca4a789277fc31f577-8080.app.kvin.wang` maps to port `8080` in the CVM.
+The `<id>` can be either the app ID or instance ID. When using the app ID, the load balancer will select one of the available instances. Adding an `s` suffix enables TLS passthrough to the app instead of terminating at dstack-gateway. Adding a `g` suffix enables HTTPS/2 with TLS termination for gRPC applications.
```

Comment:
Two questions:

- Is it only for gRPC, or also used for other http/2 applications?
- Is it necessary to have the `g` suffix?

Reply:
- It's for all HTTP/2.
- Without `g`, the gateway doesn't know HTTP/2 backend support. Falsely advertising HTTP/2 compatibility would result in the client sending incompatible packets to an HTTP/1.1 server.

Reply:
Do you think it's better to use `-h2` instead of `-g`? And for custom domain, does it detect the suffix in the TXT record?

Reply:
`*-8080-h2` would be a breaking change introducing another compatibility issue, `*-8080h2` looks a bit weird to me.

> And for custom domain, does it detect the suffix in the TXT record?

No, custom domain knows itself support h2 or not, it can only advertising h2 when the backend support it.